### PR TITLE
patch release: fix: added support for unescaped parameters in connection strings

### PR DESCRIPTION
**Pull request by tnaum-ms**
This pull request significantly improves the robustness of the `DocumentDBConnectionString` class by addressing issues with parsing MongoDB connection strings containing special characters, especially in query parameters and credentials. The changes introduce a pre-processing step to sanitize connection strings before parsing, add proper encoding/decoding for usernames, and expand test coverage to ensure correct handling of various edge cases.

**Core improvements to connection string parsing:**

* Added a static `sanitizeConnectionString` method to pre-process connection strings, ensuring special characters (such as `@`, `#`, `[`, and `]`) in query parameters are correctly encoded before being parsed by the base `ConnectionString` class. This prevents misinterpretation of credentials and hosts.

**Enhanced credential handling:**

* Overrode the `username` getter and setter to ensure usernames are properly encoded when set and decoded when retrieved, mirroring the handling already present for passwords.
* Added a static `validateUsername` method to verify that usernames can be safely encoded and decoded, preventing potential parsing errors.

**Expanded and improved testing:**

* Added comprehensive test cases to `DocumentDBConnectionString.test.ts` covering scenarios with special characters in query parameters, credentials, multiple hosts, SRV connection strings, and real-world Azure Cosmos DB and MongoDB Atlas formats. These tests validate the new parsing logic and encoding/decoding behavior.

### CODE REVIEW

File: src/documentdb/utils/DocumentDBConnectionString.test.ts [hunk 1]
```diff
@@ -163,4 +163,317 @@ describe('DocumentDBConnectionString', () => {
...
+
+            const connStr = new DocumentDBConnectionString(uri);
+
+            expect(connStr.isSRV).toBe(true);
+            // URLSearchParams.get() returns decoded values
+            expect(connStr.searchParams.get('appName')).toBe('@myapp@');
+        });
+
+        it('should handle mixed special characters in parameters', () => {
+            const uri = 'mongodb://host.example.com:27017/?tag1=@user@&tag2=[prod]&tag3=test#1';
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/documentdb/utils/DocumentDBConnectionString.test.ts]: good point. fixed.
